File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.c
#include <stdio.h>
#define SIZE 1001

int DFS_arr[SIZE];
int BFS_arr[SIZE];
int queue[SIZE];
int graph[SIZE][SIZE] = {0};

void dfs(int v, int n){
	DFS_arr[v] = 1;
	printf("%d ", v);

	for(int k=1; k<=n; k++){
		if(DFS_arr[k] == 0 && graph[v][k] == 1)
			dfs(k, n);
	}
}

void bfs(int v, int n){
	int front, rear, pop;

	front=rear=0;

	BFS_arr[v] = 1;
	queue[front] = v;	rear++;
	printf("%d ", v);
	
	while(front < rear){
		pop = queue[front];		front++;

		for(int k=1; k<=n; k++){
			if(BFS_arr[k] == 0 && graph[pop][k] == 1){
				printf("%d ", k);
				BFS_arr[k] = 1;
				queue[rear] = k;	rear++;
			}
		}
	}
}

int main(){
	int N, M, V;
	int u, v;

	scanf("%d %d %d", &N, &M, &V);

	for(int i=1; i<=M; i++){
		scanf("%d %d", &u, &v);
		graph[u][v] = 1;
		graph[v][u] = 1;
	}

	dfs(V, N);
	printf("\n");
	bfs(V, N);
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

	static int N, M, V;
	static boolean[] visited;
	static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		V = Integer.parseInt(st.nextToken());

		visited = new boolean[N + 1];

		for (int i = 0; i <= N; i++) {
			graph.add(new ArrayList<>());
		}

		for (int i = 0; i < M; i++) {
			st = new StringTokenizer(br.readLine());
			int u = Integer.parseInt(st.nextToken());
			int v = Integer.parseInt(st.nextToken());

			graph.get(u).add(v);
			graph.get(v).add(u);
		}

		for (int i = 0; i <= N; i++) {
			Collections.sort(graph.get(i));
		}

		dfs(V);
		System.out.println();

		for (int i = 0; i <= N; i++) {
			visited[i] = false;
		}

		bfs(V);
	}

	private static void dfs(int v) {
		visited[v] = true;
		System.out.print(v + " ");

		for (int i : graph.get(v)) {
			if (!visited[i]) {
				dfs(i);
			}
		}
	}

	private static void bfs(int v) {
		Queue<Integer> queue = new ArrayDeque<>();

		queue.offer(v);
		visited[v] = true;

		while (!queue.isEmpty()) {

			int num = queue.poll();
			System.out.print(num + " ");

			for (int i : graph.get(num)) {
				if (!visited[i]) {
					queue.offer(i);
					visited[i] = true;
				}
			}
		}
	}
}
